config/initializers/browse_everything.rb

```diff
@@ -11,5 +11,14 @@
   else
     settings['file_system'] = { name: 'File Dropbox', home: Settings.dropbox.path }
   end
+  if Settings.dropbox.sharepoint
+    settings['sharepoint'] = { client_id: Settings.dropbox.sharepoint.client_id,
+                               client_secret: Settings.dropbox.sharepoint.client_secret,
+                               tenant_id: Settings.dropbox.sharepoint.tenant_id,
+                               grant_type: Settings.dropbox.sharepoint.grant_type,
```

Comment:
For current implementation this line is unneeded. However, there are a couple different ways to grant an application access to Microsoft Graph. For our purposes at IU, we need to log in as specific users which requires the `authorization_code` grant type. This is currently hardcoded. The other type is for allowing the application itself to get access, so a specific user account is not needed, which uses the `client_credentials` grant type.

For the final version of this work it would be ideal to allow implementers to choose which auth flow works best for how they have sharepoint setup so I have left the `grant_type` param in place here as a placeholder/reminder.